src/fnft_nsev.c

```diff
@@ -604,37 +976,52 @@ static inline INT refine_roots_newton(
     if (T == NULL)
         return E_INVALID_ARGUMENT(T);
     
-    im_bound_val = im_bound(D, q, T);
-    if (im_bound_val == NAN)
-        return E_OTHER("Upper bound on imaginary part of bound states is NaN");
+    upsampling_factor = nse_discretization_upsampling_factor(discretization);
+    if (upsampling_factor == 0) {
+        ret_code = E_INVALID_ARGUMENT(discretization);
+        goto leave_fun;
+    }
+    D_given = D/upsampling_factor;
+    const REAL eps_t = (T[1] - T[0])/(D_given - 1);
+    
+    im_bound_val = upsampling_factor*upsampling_factor*im_bound(D, q, T);
+    if (im_bound_val == NAN){
+        ret_code = E_OTHER("Upper bound on imaginary part of bound states is NaN");
+        CHECK_RETCODE(ret_code, leave_fun);
+    }
+    re_bound_val = re_bound(eps_t, 2.0);
     
-    re_bound_val = re_bound(eps_t, discretization);
-        
     // Perform iterations of Newton's method
     for (i = 0; i < K; i++) {
         iter = 0;
         do {
             // Compute a(lam) and a'(lam) at the current root
-            ret_code = nse_scatter_bound_states(D, q, T, &trunc_index, 1,
-                bound_states + i, &a_val, &aprime_val, &b_val, discretization);
-            if (ret_code != SUCCESS)
-                return E_SUBROUTINE(ret_code);
-
+            ret_code = nse_scatter_bound_states(D, q, r, T, 1,
+                    bound_states + i, &a_val, &aprime_val, &b_val, discretization, 1);
+            if (ret_code != SUCCESS){
+                ret_code = E_SUBROUTINE(ret_code);
+                CHECK_RETCODE(ret_code, leave_fun);
+            }
             // Perform Newton updates: lam[i] <- lam[i] - a(lam[i])/a'(lam[i])
+            
+//             printf("l=%1.5e+%1.5e;\t",CREAL(bound_states[i]),CIMAG(bound_states[i]));
+//             printf("a=%1.5e+%1.5e;\t",CREAL(a_val),CIMAG(a_val));
+//             printf("aprime=%1.5e+%1.5e;\n",CREAL(aprime_val),CIMAG(aprime_val));
             if (aprime_val == 0.0)
                 return E_DIV_BY_ZERO;
             error = a_val / aprime_val;
             bound_states[i] -= error;
             iter++;
-
             if (CIMAG(bound_states[i]) > im_bound_val
```

Comment:
This might be inconsistent with the filtering method specified by the user. Should be pass the `bounding_box` that is computed anyway instead?

Reply:
Indeed that would be more consistent. I will try to pass the `bounding_box`.